src/nile/core/types/account.py

```diff
@@ -255,6 +268,13 @@ def _get_signer_and_alias(signer, predeployed_info):
     return signer, alias
 
 
+async def _set_estimated_fee_if_none(max_fee, tx):
+    """Estimate max_fee for transaction if max_fee is None."""
+    if max_fee is None:
+        estimated_fee = await tx.estimate_fee()
+        tx.update_fee(estimated_fee)
```

Comment:
```suggestion
        logger = logging.getLogger()
        current_level = logger.level

        # Avoid logging the fee estimation in CLI
        logger.setLevel(logging.WARNING)

        estimated_fee = await tx.estimate_fee()

        logger.setLevel(current_level)

        tx.update_fee(estimated_fee)
```

`nile setup ACCOUNT_1 --salt 2 --estimate_fee` is logging the fee estimation twice. And other commands are logging the fee estimation when fee is not set.

Reply:
Good catch!